lib/utils/fncache.go

```diff
@@ -182,6 +182,40 @@ func (c *FnCache) Set(key, value any) {
 	c.SetWithTTL(key, value, c.cfg.TTL)
 }
 
+// GetIfExists retrieves a value from the cache without triggering a load operation.
+// It returns (value, true) if a valid, non-expired entry exists, or (nil, false)
+// otherwise. If an entry is currently being loaded by FnCacheGet, Get will
+// return false immediately without blocking. Get returns false for entries that
+// contain errors.
+// For most of the cases the FnCacheGet function should be used instead.
+func (c *FnCache) GetIfExists(key any) (any, bool) {
```

Comment:
Can we make this a free function similar to `FnCacheGet` instead of a receiver so that we can leverage generics and eliminate the need to cast from an any by the caller?
```suggestion
func FnCacheGetIfExists[K comparable, T any](cache *FnCache, key K) (T, bool) {
```

Reply:
If can find many articles  how GO generic  types are deduced. In this case it is not possible so 

```
 FnCacheGetIfExists(cache, "test-key")
 ```

gives compiler error

` in call to FnCacheGetIfExists, cannot infer T (declared at ./fncache.go:185:39) `

So you need to call `FnCacheGetIfExists[string, string](cache, "test-key")`   with explicit types provided `[string, string]`   that is very ugly and obscure for the usage and is quite easy to misused the types

Reply:
Why is this not the case for existing calls to FnCacheGet?

https://github.com/search?q=repo%3Agravitational%2Fteleport%20utils.FnCacheGet&type=code

Reply:
Please see https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#function-argument-type-inference

Go complier can't infer function type arguments from function like `func FnCacheGetIfExists[K comparable, T any](cache *FnCache, key K) (T, bool) {`

where FnCacheGetIfExists and FnCacheGet have totally diffrent signature.